src/components/Airtable.js

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import {
+  InternalLinks,
+  AIRTABLE_ASSETS_RULES,
+  AIRTABLE_CATEGORIES_RULES,
+  PULL_ASSETS_TIME,
+  PULL_CATEGORIES_TIME,
+  verifyDataAirtable,
+} from '../constants';
+
+export const AirtableContext = React.createContext(null);
+
+class AirtableProvider extends React.Component {
+  constructor(props){
+    super(props);
+    this.getCategoriesForAssets = this.getCategoriesForAssets;
+    this.state = {
+      getCategoriesForAssets: this.getCategoriesForAssets,
+    }
+  }
+
+  componentDidMount = () => {
+    this.getAssets();
+    this.getCategories();
+    this.setIntervals();
+  }
+
+  componentWillUnmount = () => {
+    this.resetIntervals();
+  }
+
+  setIntervals = () => {
+    this.intervalPullAssets = setInterval(this.getAssets, PULL_ASSETS_TIME)
+    this.intervalPullCategories = setInterval(this.getCategories, PULL_CATEGORIES_TIME)
+  }
+
+  resetIntervals = () => {
+    clearInterval(this.intervalPullAssets);
+    clearInterval(this.intervalPullCategories);
+  }
+
+  processAssetsFromAirTable = ({ fields }) => {
+    let location = undefined;
+    if(fields.Location){
+      let countries = fields.Location.split(',');
+      location = {};
+      countries.forEach(country => {
+        country = country.trim();
+        let cities = /\(([^)]+)\)/g.exec(country);
+
+        if(cities){
+          country = country.substring(0, country.indexOf('('))
+          cities = cities[1].split(';');
+          location[country] = cities;
+        } else {
+           location[country] = {};
+        }
+      })
+    }
+
+    return {
+      name: fields.Asset,
+      category: fields.Category,
+      description: fields.Description,
+      details: fields.Details,
+      partner: fields.Partner,
+      imageSrc: `${InternalLinks.S3}assetImages:${fields.Image}`,
+      amountToBeRaisedInUSDAirtable: fields['Funding goal'],
+      assetIDs: fields['Asset IDs'],
+      location,
+    };
+  }
+
+  processCategoriesFromAirTable = (data) => {
+    const categories = {};
+    data.forEach(({ fields }) => {
+      categories[fields.Category] = {
+        contractName: fields['Category Contract'],
+        encoded: fields['byte32'],
+      }
+    })
+    return categories;
+  }
+
+  getAssetByName = (assetName, assetsFromAirTable) => {
+    const tmpAsset = Object.assign({}, assetsFromAirTable.filter(asset => asset.name === assetName)[0]);
+    tmpAsset.location = undefined;
```

Comment:
Might as well do `delete tmpAsset.location`. Also, the previous statement assumes you are only retrieve one object, but if you don't, you return an empty `object`. No need to do `Object.assign`, you can just do
`assetsFromAirTable.filter(asset => asset.name === assetName)[0] || {}`